sdk/src/mesh_dp.cc

```diff
@@ -0,0 +1,268 @@
+
+    ConnectionContext *conn_ctx = (ConnectionContext *)conn;
+
+    return conn_ctx->apply_config_audio(cfg);
+}
+
+/**
+ * Establish a mesh connection
+ */
+int mesh_establish_connection(MeshConnection *conn, int kind)
+{
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    ConnectionContext *conn_ctx = (ConnectionContext *)conn;
+
+    return conn_ctx->establish(kind);
+}
+
+/**
+ * Shutdown a mesh connection
+ */
+int mesh_shutdown_connection(MeshConnection *conn)
+{
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    ConnectionContext *conn_ctx = (ConnectionContext *)conn;
+
+    return conn_ctx->shutdown();
+}
+
+/**
+ * Delete mesh connection
+ */
+int mesh_delete_connection(MeshConnection **conn)
+{
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    ConnectionContext *conn_ctx = (ConnectionContext *)(*conn);
+
+    if (!conn_ctx)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    int err = conn_ctx->shutdown();
+    if (err)
+        return err;
+
+    if (!conn_ctx->__public.client)
+        return -MESH_ERR_BAD_CLIENT_PTR;
+
+    delete conn_ctx;
+    *conn = NULL;
+
+    return 0;
+}
+
+/**
+ * Get buffer from mesh connection
+ */
+int mesh_get_buffer(MeshConnection *conn, MeshBuffer **buf)
+{
+    return mesh_get_buffer_timeout(conn, buf, MESH_TIMEOUT_DEFAULT);
+}
+
+/**
+ * Get buffer from mesh connection with timeout
+ */
+int mesh_get_buffer_timeout(MeshConnection *conn, MeshBuffer **buf,
+                            int timeout_ms)
+{
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    ConnectionContext *conn_ctx = (ConnectionContext *)conn;
+
+    return conn_ctx->get_buffer_timeout(buf, timeout_ms);
+}
+
+/**
+ * Put buffer to mesh connection
+ */
+int mesh_put_buffer(MeshBuffer **buf)
+{
+    return mesh_put_buffer_timeout(buf, MESH_TIMEOUT_DEFAULT);
+}
+
+/**
+ * Put buffer to mesh connection with timeout
+ */
+int mesh_put_buffer_timeout(MeshBuffer **buf, int timeout_ms)
+{
+    if (!buf)
+        return -MESH_ERR_BAD_BUF_PTR;
+
+    BufferContext *buf_ctx = (BufferContext *)(*buf);
+
+    if (!buf_ctx)
+        return -MESH_ERR_BAD_BUF_PTR;
+
+    int err = buf_ctx->enqueue(timeout_ms);
+    if (err)
+        return err;
+
+    delete buf_ctx;
+    *buf = NULL;
+
+    return 0;
+}
+
+/**
+ * Get text description of an error code.
+ */
+const char *mesh_err2str(int err)
+{
+    switch (err) {
+    case -MESH_ERR_BAD_CLIENT_PTR:
+        return "Bad client pointer";
+    case -MESH_ERR_BAD_CONN_PTR:
+        return "Bad connection pointer";
+    case -MESH_ERR_BAD_CONFIG_PTR:
+        return "Bad configuration pointer";
+    case -MESH_ERR_BAD_BUF_PTR:
+        return "Bad buffer pointer";
+    case -MESH_ERR_MAX_CONN:
+        return "Reached max number of connections";
+    case -MESH_ERR_FOUND_ALLOCATED:
+        return "Found allocated resources";
+    case -MESH_ERR_CONN_FAILED:
+        return "Connection creation failed";
+    case -MESH_ERR_CONN_CONFIG_INVAL:
```

Comment:
This err is used in many different config error types, I understand that they all are config errors, but perhaps adding specific errors for audio/transport/video configs could be beneficial for quick debugging?
I remeber the case when openssl threw general config file error, took me a lot of debugging to rootcause that to permisions on HW QAT device access rights, because of error info generalisation in openssl. Just a thought...

Reply:
Good idea. It should be for the future effort, not for this PR.